jujugui/static/gui/src/app/store/env/base.js

```diff
@@ -337,12 +282,11 @@ YUI.add('juju-env-base', function(Y) {
       // Consider the user unauthenticated until proven otherwise.
       this.userIsAuthenticated = false;
       this.failedAuthentication = false;
-      // Populate our credentials if they don't already exist.
-      const credentials = this.getCredentials();
+      const credentials = this.get('user').controller;
       if (!credentials.areAvailable) {
-        credentials.user = this.get('user') || '';
-        credentials.password = this.get('password') || '';
-        this.setCredentials(credentials);
+        credentials.user = '';
+        credentials.password = '';
+        this.get('user').controller = credentials;
```

Comment:
I am not sure 285-289 is necessary as the `user.controller` will return empty strings for the `user` and `password` fields if none are provided anyways.